man/man3/lzc_set_props.3

```diff
@@ -0,0 +1,208 @@
+.\" You may not use this file except in compliance with the License.
+.\"
+.\" You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+.\" or http://www.opensolaris.org/os/licensing.
+.\" See the License for the specific language governing permissions
+.\" and limitations under the License.
+.\"
+.\" When distributing Covered Code, include this CDDL HEADER in each
+.\" file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+.\" If applicable, add the following below this CDDL HEADER, with the
+.\" fields enclosed by brackets "[]" replaced with your own identifying
+.\" information: Portions Copyright [yyyy] [name of copyright owner]
+.\"
+.\" CDDL HEADER END
+.\"
+.\"
+.\" Copyright 2015 ClusterHQ Inc. All rights reserved.
+.\"
+.TH lzc_set_props 3 "2015 JUL 7" "OpenZFS" "OpenZFS Programmer's Manual"
+
+.SH PROLOG
+This manual page is part of the OpenZFS Programmer's Manual.
+
+.SH SYNOPSIS
+#include <libzfs_core.h>
+
+\fBint\fR \fBlzc_set_props\fR(\fBconst char *\fR\fIfsname\fR, \fBnvlist_t *\fR\fIprops\fR, \fBboolean_t\fR \fIreceived\fR);
+
+.SH NAME
+lzc_set_props \- Set properties on a filesystem or volume
+
+.SH DESCRIPTION
+.LP
+The \fBlzc_set_props\fR() function will set properties on a given dataset or volume.
+.sp
+.I fsname
+is a string containing the name of a dataset or filesystem.
+.sp
+.I props
+is a nvlist containing the properties to set. See \fBzfs\fR(8) for descriptions
+of properties.
+.sp
+.I opts
+is a nvlist containing additional options to pass to this command.
+.sp
+The \fIprops\fR parameter is an nvlist of property names
+(with no values) that will be returned for each bookmark. The \fIbmarks\fR
+parameter is a pointer to a nvlist of the bookmarks. The caller is responsible
+for freeing it with \fBnvlist_free\fR().
+
+The format of the returned nvlist as follows:
+.P
+<short name of bookmark> -> {
+    <name of property> -> {
+        "value" -> uint64
+     }
+.br
+}
+
+The following are valid properties on bookmarks. All of them are uint64 values
+in the nvlist:
+.sp
+.na
+\fB\fBguid\fR\fR
+.ad
+.RS
+Globally unique identifier of the referenced snapshot
+.RE
+
+.na
+\fB\fBcreatetxg\fR\fR
+.ad
+.RS
+Number of transaction group (txg) when the snapshot provided to the bookmark was created
+.RE
+
+.na
+\fB\fBcreation\fR\fR
+.RS
+UNIX timestamp when the snapshot used to make the bookmark was created
+.RE
+.sp
+ Currently, the \fIopts\fR argument has two additional options.
+.sp
+.I received
+A boolean property that specifies that the properties being set were received from a send stream.
+.sp
+.I noatomic
+tells the kernel to handle failures by setting all properties that do not fail.
+This is necessary when setting old-style special properties.
+.sp
+There are 7 old-style special properties that cannot be set atomically with the others. They are:
+.sp
+groupquota@
+.sp
+groupused@
+.sp
+snapdev
+.sp
+userquota@
+.sp
+userused@
+.sp
+version
+.sp
+volsize
```

Comment:
filesystem_limit and snapshot_limit were developed concurrently with the atomic behavior and are technically old-style. They need to be documented.